### PR TITLE
[Merged by Bors] - feat: port Analysis.LocallyConvex.Bounded

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
bors merge

File: Mathlib/Analysis/LocallyConvex/Bounded.lean [hunk 1]
```diff
@@ -170,25 +173,26 @@ theorem isVonNBounded_of_smul_tendsto_zero {ε : ι → 𝕝} {l : Filter ι} [l
...
   indexing type `ι`, but in the special case `ι = ℕ` we get the important fact that convergent
   sequences fully characterize bounded sets. -/
-theorem isVonNBounded_iff_smul_tendsto_zero {ε : ι → 𝕝} {l : Filter ι} [l.ne_bot]
+theorem isVonNBounded_iff_smul_tendsto_zero {ε : ι → 𝕝} {l : Filter ι} [l.NeBot]
     (hε : Tendsto ε l (𝓝[≠] 0)) {S : Set E} :
     IsVonNBounded 𝕝 S ↔ ∀ x : ι → E, (∀ n, x n ∈ S) → Tendsto (ε • x) l (𝓝 0) :=
   ⟨fun hS x hxS => hS.smul_tendsto_zero (eventually_of_forall hxS) (le_trans hε nhdsWithin_le_nhds),
-    isVonNBounded_of_smul_tendsto_zero (hε self_mem_nhdsWithin)⟩
+    -- Porting note: was `isVonNBounded_of_smul_tendsto_zero (hε self_mem_nhdsWithin)`
+    by apply isVonNBounded_of_smul_tendsto_zero; exact hε self_mem_nhdsWithin⟩
```
Comment (eric-wieser) [inline Mathlib/Analysis/LocallyConvex/Bounded.lean]: This one is a little weird; does `(isVonNBounded_of_smul_tendsto_zero (hε self_mem_nhdsWithin) :)` or `isVonNBounded_of_smul_tendsto_zero  (by exact hε self_mem_nhdsWith)` work?

Comment (Parcly-Taxel) [reply Mathlib/Analysis/LocallyConvex/Bounded.lean]: The latter works @eric-wieser; I've just edited that in.

File: Mathlib/Analysis/LocallyConvex/Bounded.lean [hunk 2]
```diff
@@ -151,15 +154,15 @@ theorem IsVonNBounded.smul_tendsto_zero {S : Set E} {ε : ι → 𝕜} {x : ι 
   rw [tendsto_def] at *
   intro V hV
   rcases hS hV with ⟨r, r_pos, hrS⟩
-  filter_upwards [hxS, hε _ (Metric.ball_mem_nhds 0 <| inv_pos.mpr r_pos)]with n hnS hnr
-  by_cases this : ε n = 0
-  · simp [this, mem_of_mem_nhds hV]
-  · rw [mem_preimage, mem_ball_zero_iff, lt_inv (norm_pos_iff.mpr this) r_pos, ← norm_inv] at hnr
-    rw [mem_preimage, Pi.smul_apply', ← Set.mem_inv_smul_set_iff₀ this]
+  filter_upwards [hxS, hε _ (Metric.ball_mem_nhds 0 <| inv_pos.mpr r_pos)] with n hnS hnr
+  by_cases dis : ε n = 0
```
[the remark this answers is not in the document]
Comment (Parcly-Taxel) [reply Mathlib/Analysis/LocallyConvex/Bounded.lean]: Done.

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**